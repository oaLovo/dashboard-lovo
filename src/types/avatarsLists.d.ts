export type ListDataProps = {
  avatar?: string
  name: string
  subtitle: string
}

export interface AvatarsListsProps {
  listData: ListDataProps[]
}
