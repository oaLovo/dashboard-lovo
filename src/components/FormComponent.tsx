import styled from 'styled-components'
import { FormComponentProps } from '@/types'
import { StyledButton, StyledInput, StyledP } from '@/components'
import { pxToRem } from '@/utils'

const StyledForm = styled.form`
  display: flex;
  flex-direction: column;
  row-gap: ${pxToRem(16)};
`

function FormComponent(props: FormComponentProps) {
  const { inputs, buttons, message } = props
  return (
    <StyledForm>
      {inputs.map((inputProps, index) => (
        <StyledInput key={index} {...inputProps} />
      ))}
      {buttons.map((buttonProps, index) => (
        <StyledButton key={index} {...buttonProps} />
      ))}
      {message && (
        <StyledP className={message.type === 'error' ? 'error' : 'success'}>
          {message.msg}
        </StyledP>
      )}
    </StyledForm>
  )
}

export default FormComponent
