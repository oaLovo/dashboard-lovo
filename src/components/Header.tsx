import styled from 'styled-components'
import { Link } from 'react-router-dom'

import { Logo } from '@/components'

// MUI
import Avatar from '@mui/material/Avatar'
import Box from '@mui/material/Box'
import Container from '@mui/material/Container'

// UTILS
import { pxToRem } from '@/utils'

const StyledHeader = styled.header`
  background-color: ${(props) => props.theme.appBackground};
  border-bottom: ${pxToRem(1)} solid ${(props) => props.theme.appDefaultStroke};
  margin-bottom: ${pxToRem(37)};
  width: 100%;
`

function Header() {
  return (
    <StyledHeader>
      <Container maxWidth="lg">
        <Box
          sx={{
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'space-between',
            height: pxToRem(64),
          }}
        >
          <Link to="/home">
            <Logo height={30} width={73} />
          </Link>
          <Link to="/perfil">
            <Avatar
              alt="DNC Avatar"
              src="/dnc-avatar.jpg"
              sx={{ width: pxToRem(40), height: pxToRem(40) }}
            />
          </Link>
        </Box>
      </Container>
    </StyledHeader>
  )
}

export default Header
