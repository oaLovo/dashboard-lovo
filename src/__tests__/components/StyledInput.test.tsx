import '@testing-library/jest-dom'
import 'jest-styled-components'
import { pxToRem } from '@/utils'
import { render } from '@testing-library/react'
import { StyledInput } from '@/components'
import { Theme } from '@/types'
import { ThemeProvider } from 'styled-components'
import { themesList } from '@/resources/themesList'

describe('StyledInput Component', () => {
  const renderComponent = (theme: Theme, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledInput {...props} />
      </ThemeProvider>
    )

  themesList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
      it('should match snapshot', () => {
        const { asFragment } = renderComponent(theme)
        expect(asFragment()).toMatchSnapshot()
      })

      it('should apply active styles', () => {
        const { getByRole } = renderComponent(theme)
        const input = getByRole('textbox')
        expect(input).toHaveStyleRule(
          'background-color',
          theme.textInput.active
        )
        expect(input).toHaveStyleRule('color', theme.textInput.activeColor)
        expect(input).toHaveStyleRule(
          'border',
          `${pxToRem(1)} solid ${theme.textInput.borderColor}`
        )
      })

      it('should apply disabled styles', () => {
        const { getByRole } = renderComponent(theme, { disabled: true })
        const input = getByRole('textbox')
        expect(input).toHaveStyleRule(
          'background-color',
          theme.textInput.disabled,
          { modifier: ':disabled' }
        )
        expect(input).toHaveStyleRule(
          'border',
          `${pxToRem(1)} solid ${theme.textInput.disabledBorderColor}`,
          { modifier: ':disabled' }
        )
        expect(input).toHaveStyleRule('color', theme.textInput.disabledColor, {
          modifier: ':disabled',
        })
      })

      it('should apply placeholder styles', () => {
        const { getByPlaceholderText } = renderComponent(theme, {
          placeholder: 'Type here...',
        })
        const input = getByPlaceholderText('Type here...')
        expect(input).toHaveStyleRule(
          'color',
          theme.textInput.placeholderColor,
          { modifier: '::placeholder' }
        )
      })
    })
  })
})
