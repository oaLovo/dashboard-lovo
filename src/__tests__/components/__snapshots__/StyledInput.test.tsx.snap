// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledInput Component Dark Theme should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #0F1535;
  color: #FFF;
  border-radius: 0.5rem;
  border: 0.0625rem solid #21497D;
  box-sizing: border-box;
  cursor: pointer;
  height: 2.5rem;
  font-size: 0.875rem;
  font-weight: 500;
  padding: 0.5rem 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0:disabled {
  background-color: #282D49;
  border: 0.0625rem solid #2E3F55;
  color: #58677C;
  cursor: not-allowed;
}

.c0::placeholder {
  color: #89A7CE;
}

<input
    class="c0"
  />
</DocumentFragment>
`;

exports[`StyledInput Component Light Theme should match snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #FFF;
  color: #000;
  border-radius: 0.5rem;
  border: 0.0625rem solid #E0E0E0;
  box-sizing: border-box;
  cursor: pointer;
  height: 2.5rem;
  font-size: 0.875rem;
  font-weight: 500;
  padding: 0.5rem 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0:disabled {
  background-color: #EEE;
  border: 0.0625rem solid #E0E0E0;
  color: #666;
  cursor: not-allowed;
}

.c0::placeholder {
  color: #666;
}

<input
    class="c0"
  />
</DocumentFragment>
`;
